frontend/vre/utils/jsonld.model.js

```diff
@@ -1,11 +1,91 @@
 import {APICollection} from "./api.model";
 
 /**
- * Generic subclass of APICollection that parses incoming JSON-LD to a
- * list of subjects.
+ * The @graph property inside JSON-LD
+ * @typedef {Object} JSONLDGraph
+ */
+
+/**
+ * An individual subject definition inside JSON-LD
+ * @typedef {Object} JSONLDSubject
+ */
+
+/**
+ * Generic subclass of APICollection that parses incoming JSON-LD to an
+ * array of all subjects. The contents of subjects are left unchanged.
  */
 export var JsonLdCollection = APICollection.extend({
     parse: function(response) {
+        if (!response.hasOwnProperty("@graph")) {
+            throw "Response has no @graph key";
+        }
```

Comment:
Not now, because the backend always adds a context and returns the compacted form. I have added some notes that this compacted form is expected. However, I think that some additional robustness is needed, because I have only tested the function with data coming from rdflib... I have created issue #184 for this.